### PR TITLE
feat: get user streak

### PR DESCRIPTION
**Pull request by sshanzel**
Introduce a new query to fetch the user's reading streak.

Note: this PR is purely retrieval. The additional logic that resets the user's current streak based on some checks will be for a different ticket/PR.

MI-72 #done

### CODE REVIEW

File: src/graphorm/index.ts [hunk 1]
```diff
@@ -74,6 +74,13 @@ const obj = new GraphORM({
   CommentUpvote: {
     requiredColumns: ['createdAt'],
   },
+  UserStreak: {
+    fields: {
+      max: { select: 'maxStreak' },
+      total: { select: 'totalStreak' },
+      current: { select: 'currentStreak' },
```
Comment (idoshamun) [inline src/graphorm/index.ts:81]: Why do we have separate names for the db columns and the graphql ones? Why don't we keep the same name?

Comment (sshanzel) [reply src/graphorm/index.ts:81]: Took it as how it was desired from the DR. But we can probably keep the same names to reduce complexity. cc @DragosIorgulescu wdyt?

**Review by idoshamun — APPROVED** [review]
I'm approving but just double check with Dragos. I'm ok either way

File: src/graphorm/index.ts (hunk 1 above)
Comment (rebelchris) [reply src/graphorm/index.ts:81]: Yeah you query `userStreak.something` right?
Would make sense to me to keep them inline.

Comment (DragosIorgulescu) [reply src/graphorm/index.ts:81]: we can change the naming here to be the same, up to you if you want to use single word variations @sshanzel

**Review by DragosIorgulescu — COMMENTED** [review]
> Note: this PR is purely retrieval. The additional logic that resets the user's current streak based on some checks will be for a different ticket/PR.

Could you point to the ticket where this would be covered? I think the ticket for this includes the clearing of the `currentStreak` functionality as well.

File: src/graphorm/index.ts (hunk 1 above)
Comment (sshanzel) [reply src/graphorm/index.ts:81]: Alright. Let's do single-word names since we also had that on our RFC too 🥳

**Comment by sshanzel** [comment]
> Could you point to the ticket where this would be covered? I think the ticket for this includes the clearing of the currentStreak functionality as well.

This PR is for the MI-72 ticket. I've thought of that as well. But then there is a separate ticket specifically for that (modifying this query). https://dailydotdev.atlassian.net/jira/software/projects/MI/boards/31/backlog?epics=visible&issueParent=14432&selectedIssue=MI-108

Which is why I also had to ask it in the [relevant ticket](https://dailydotdev.atlassian.net/browse/MI-72?focusedCommentId=11331).

**Comment by DragosIorgulescu** [comment]
> > Could you point to the ticket where this would be covered? I think the ticket for this includes the clearing of the currentStreak functionality as well.
> 
> This PR is for the [MI-72](https://dailydotdev.atlassian.net/jira/software/projects/MI/boards/31?selectedIssue=MI-72) ticket. I've thought of that as well. But then there is a separate ticket specifically for that (modifying this query). https://dailydotdev.atlassian.net/jira/software/projects/MI/boards/31/backlog?epics=visible&issueParent=14432&selectedIssue=MI-108
> 
> Which is why I also had to ask it in the [relevant ticket](https://dailydotdev.atlassian.net/browse/MI-72?focusedCommentId=11331).

@sshanzel you're right, I created a separate ticket for that (and forgot I did 🤦‍♂️ ) -- we will tackle it separately, this PR should be good to go 🙏

**Review by DragosIorgulescu — APPROVED** [review]
🚀

**Pull request merged**